src/App.tsx

```diff
@@ -15,36 +15,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetic = 'alphabetic',
+  Length = 'length',
+  Reset = 'reset',
+  Reverse = 'reverse',
+}
+
+function sortingElement(
+  goods: string[],
+  sortField: SortType,
+  isReversed: boolean,
+): string[] {
+  const sortedGoods = [...goods];
+
+  switch (sortField) {
+    case SortType.Alphabetic:
+      sortedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+    case SortType.Length:
+      sortedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+    case SortType.Reset:
+      return goodsFromServer;
+    default:
+      break;
+  }
+
+  return isReversed ? sortedGoods.reverse() : sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>();
+  const [isReversed, setisReversed] = useState(false);
+
+  const goods = sortingElement(
+    goodsFromServer,
+    sortField as SortType,
+    isReversed,
+  );
+
+  const handleSort = (field: SortType) => {
+    if (field === SortType.Reverse) {
+      setisReversed(!isReversed);
+    } else {
+      setSortField(field);
+    }
+  };
+
+  const handleReset = () => {
+    setSortField('');
```

Comment:
The `setSortField` function is being called with an empty string, which is not compatible with the `SortType` enum. Consider using a valid `SortType` value, such as `SortType.Reset`, to reset the sorting.